solr/solr-ref-guide/src/the-tagger-handler.adoc

```diff
@@ -271,11 +271,12 @@ The response should be this (the QTime may vary):
   }}
 ----
 
-== Tagger Tips
+== Tagger Performance Tips
 
-Performance Tips:
-
-* Follow the recommended configuration field settings, especially `postingsFormat=FST50`.
+* Follow the recommended configuration field settings above.
+Additionally, for the best tagger performance, set `postingsFormat=FST50`.
+However, non-default postings formats have no backwards-compatibility guarantees, and so if you upgrade Solr then you may find a nasty exception on startup as it fails to read the older index.
+If the input text to be tagged is small (e.g. you are tagging queries or tweets) then the postings format choice isn't as important.
```

Comment:
[Q] Interesting.  I didn't realize that the FST50 vs default performance decreased the smaller the individual document size was.  Did you do a particular performance test to bear this out, or are you just intuiting that behavior from knowing how postingsFormats work?

Is the performance comparable even if numTweets or whatever gets large and the posting-lists grow due to the sheer number of tiny docs?

Reply:
> I didn't realize that the FST50 vs default performance decreased the smaller the individual document size was

The tagger works by looping over each token from the input and doing a term dictionary lookup on the local index.  Logically, if your input text is small then there is less work to do than for large input text.  Knowing this requires tagger knowledge but not how any particular postings format works.  See?  No I didn't benchmark this ;-).

Reply:
FYI the SolrTextTagger was benchmarked a couple years ago to compare the old "Memory" PF and FST50 -- https://github.com/OpenSextant/SolrTextTagger/issues/38#issuecomment-385597248   we never tried the default (blocktree).  I believe the input data in that experiment were whole articles, and thus would be impacted by the postings format choice.

Reply:
Makes sense.

I ran into that old perf-test comment back when I was raising SOLR-14254, but wasn't sure how relevant it was.  The situation user's face today is very different: "memory" (the most performant option) is gone entirely and the test doesn't even try to cover blocktree.

But that's independent of this PR.  These docs are worth getting in as-is.  If one of us (or someone else entirely) is able to shed more light on blocktree performance in the future, docs can be updated at that point.  No reason to let the perfect get in the way of the good.